certified-connectors/DocuSign/script.csx

```diff
@@ -1471,13 +1464,46 @@ public class Script : ScriptBase
   private string GetEnvelopeUrl(JToken envelope)
   {
     var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
-    var envelopeUrl = uriBuilder.Uri.ToString().Contains("demo") ?
-      "https://apps-d.docusign.com/send/documents/details/" + envelope["envelopeId"] :
-      "https://app.docusign.com/documents/details/" + envelope["envelopeId"];
+    var envelopePath = uriBuilder.Uri.ToString().Contains("demo") ?
+        "/send/documents/details/" + envelope["envelopeId"]
+      : uriBuilder.Uri.ToString().Contains(".mil") ?
+        "/documents/details/" + envelope["envelopeId"]
+      : uriBuilder.Uri.ToString().Contains("stage") ?
+        "/send/documents/details/" + envelope["envelopeId"]
+      : "/documents/details/" + envelope["envelopeId"];
+    var envelopeUrl = GetApplicationURL() + envelopePath;
 
     return envelopeUrl;
   }
 
+  private string GetApplicationURL()
+  {
+    var host = this.Context.Request.RequestUri.Host.ToLower();
+    var applicationURL = host.Contains("demo") ?
+        "https://apps-d.docusign.com"
+      : host.Contains("stage") ?
+        "https://apps-s.docusign.com"
+      : host.Contains(".mil") ?
+        "https://app.docusign.mil"
+      : "https://app.docusign.com";
+
+    return applicationURL;
+  }
+
+  private string GetHostURL()
```

Comment:
consider renaming to `GetAccountServerBaseUri`

Reply:
Done